src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -332,12 +332,37 @@ private int commonPrefixLength(List<String> classNames) {
     for (int i = 0; i < size; i++) {
       String a = classNames.get(i);
       String b = typeSpecStack.get(i).name;
-      if (!a.equals(b)) return i;
+      if (!a.equals(b)) {
```

Comment:
I think we need another method. This method no longer does what it's named to do, nor what it's documented to do. That new method should delegate to `commonPrefixLength()`.